pallets/rc-migrator/src/vesting.rs

```diff
@@ -94,3 +95,65 @@ impl<T: Config> PalletMigration for VestingMigrator<T> {
 		Ok(inner_key)
 	}
 }
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct GenericVestingInfo<BlockNumber, Balance> {
+	pub locked: Balance,
+	pub starting_block: BlockNumber,
+	pub per_block: Balance,
+}
+
+#[cfg(feature = "std")]
+impl<T: Config> crate::types::RcMigrationCheck for VestingMigrator<T> {
+	type RcPrePayload =
+		Vec<(Vec<u8>, Vec<BalanceOf<T>>, Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>>)>;
+
+	fn pre_check() -> Self::RcPrePayload {
+		pallet_vesting::Vesting::<T>::iter()
+			.map(|(who, schedules)| {
+				let balances: Vec<BalanceOf<T>> = schedules.iter().map(|s| s.locked()).collect();
+				let vesting_info: Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>> =
+					schedules
+						.iter()
+						.map(|s| GenericVestingInfo {
+							locked: s.locked(),
+							starting_block: s.starting_block(),
+							per_block: s.per_block(),
+						})
+						.collect();
+				(who.encode(), balances, vesting_info)
+			})
+			.collect()
+	}
+
+	fn post_check(pre_payload: Self::RcPrePayload) {
+		let pre_map: BTreeMap<
+			Vec<u8>,
+			(Vec<BalanceOf<T>>, Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>>),
+		> = pre_payload
+			.into_iter()
+			.map(|(who, balances, vesting_info)| (who, (balances, vesting_info)))
+			.collect();
+
+		let current_map: BTreeMap<
+			Vec<u8>,
+			(Vec<BalanceOf<T>>, Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>>),
+		> = pallet_vesting::Vesting::<T>::iter()
+			.map(|(who, schedules)| {
+				let balances: Vec<BalanceOf<T>> = schedules.iter().map(|s| s.locked()).collect();
+				let vesting_info: Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>> =
+					schedules
+						.iter()
+						.map(|s| GenericVestingInfo {
+							locked: s.locked(),
+							starting_block: s.starting_block(),
+							per_block: s.per_block(),
+						})
+						.collect();
+				(who.encode(), (balances, vesting_info))
+			})
+			.collect();
+
+		assert_eq!(pre_map, current_map, "Vesting data mismatch after migration");
```

Comment:
Is this just the same test as in the AH post check?! It should probably check that the data is all gone from the relay.  
Although there could be a bug with the data being removed, hence why we need these tests.  
But it does not make sense to test the same thing twice.

Reply:
My idea was to check that the vesting currently present in RC is equal to RcPrePayload. It's a sort of double check to verify that the migrated RcPrePayload is complete and that there were no issues beforehand, making it slightly different from the one in AH. I could be completely wrong, but looking at the vesting migration, it didn't seem like the vesting was being removed, which is why I didn't check if it was empty.

Reply:
we probably just forgot to remove the items from the storage on RC. @ggwpez should tell for sure. other than that this PR look good to me.